reader.go

```diff
@@ -297,7 +310,7 @@ func (r *CBORReader) ReadIntArray() ([]int, error) {
 	return out, nil
 }
 
-func (r *CBORReader) ReadStringMap() (map[string]interface{}, error) {
+func (r *CBORReader) ReadStringMap() (map[string]TaggedElement, error) {
```

Comment:
The corresponding `WriteStringMap` should mirror this same interface, and provide a utility method `WriteStringMapUntagged` which is a convenience method for passing in a normal map.